server/src/main/java/org/elasticsearch/index/mapper/NestedPathFieldMapper.java

```diff
@@ -0,0 +1,172 @@
+                                                      ParserContext parserContext) throws MapperParsingException {
+            throw new MapperParsingException(name(parserContext.mapperService().getIndexSettings().getSettings()) + " is not configurable");
+        }
+
+        @Override
+        public MetadataFieldMapper getDefault(MappedFieldType fieldType, ParserContext context) {
+            final IndexSettings indexSettings = context.mapperService().getIndexSettings();
+            return new NestedPathFieldMapper(indexSettings, fieldType);
+        }
+    }
+
+    public static final class NestedPathFieldType extends StringFieldType {
+
+        NestedPathFieldType() {
+        }
+
+        protected NestedPathFieldType(NestedPathFieldType ref) {
+            super(ref);
+        }
+
+        @Override
+        public MappedFieldType clone() {
+            return new NestedPathFieldType(this);
+        }
+
+        @Override
+        public String typeName() {
+            return NAME;
+        }
+
+        @Override
+        public boolean isSearchable() {
+            return true;
+        }
+
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            throw new UnsupportedOperationException("Cannot run exists() query against the nested field path");
+        }
+    }
+
+    private NestedPathFieldMapper(IndexSettings indexSettings, MappedFieldType existing) {
+        this(existing == null ? defaultFieldType(indexSettings) : existing.clone(),
+            indexSettings);
+    }
+
+    private NestedPathFieldMapper(MappedFieldType fieldType, IndexSettings indexSettings) {
+        super(name(indexSettings.getSettings()), fieldType, defaultFieldType(indexSettings), indexSettings.getSettings());
+    }
+
+    private static MappedFieldType defaultFieldType(IndexSettings indexSettings) {
+        MappedFieldType defaultFieldType = Defaults.FIELD_TYPE.clone();
+        defaultFieldType.setIndexOptions(IndexOptions.NONE);
+        defaultFieldType.setHasDocValues(false);
+        defaultFieldType.setName(name(indexSettings.getSettings()));
+        return defaultFieldType;
+    }
+
+    @Override
+    public void preParse(ParseContext context) throws IOException {
+        super.parse(context);
+    }
+
+    @Override
+    public void parse(ParseContext context) throws IOException {
+        // we parse in pre parse
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context, List<IndexableField> fields) throws IOException {
+        if (fieldType().indexOptions() == IndexOptions.NONE && !fieldType().stored()) {
+            return;
+        }
+        fields.add(new Field(fieldType().name(), MapperService.SINGLE_MAPPING_NAME, fieldType()));
+        if (fieldType().hasDocValues()) {
+            fields.add(new SortedSetDocValuesField(fieldType().name(), new BytesRef(MapperService.SINGLE_MAPPING_NAME)));
+        }
```

Comment:
I don't think we need to index root document in this field. We use the nested path to find nested documents but we don't use this field to detect root documents so this indexing would be redundant ?

Reply:
++ I pushed 7ea0da8